app/src/main/java/com/perflyst/twire/fragments/StreamFragment.java

```diff
@@ -376,6 +386,74 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+            } else {
+                Log.d("Runtime String", args.getString(getString(R.string.stream_fragment_runtime)));
+                Date date = null;
+                SimpleDateFormat format = new SimpleDateFormat("HH:mm:ss");
+                try {
+                    date = format.parse(String.valueOf(args.getString(getString(R.string.stream_fragment_runtime))));
+                    Log.d("Date", String.valueOf(date));
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+
+                int startminute = date.getMinutes();
+                int starthour = date.getHours();
+
+                // https://stackoverflow.com/questions/46774195/countdown-reverse-counter
+                // this is a reverse countdowntimer
+                new CountDownTimer(999999999, 1000) {
+                    int min = startminute;
+                    int hour = starthour;
+
+                    public void onTick(long millisUntilFinished) {
+                        long time = 999999999 - millisUntilFinished;
+                        long sec = (time/1000)%60;
+                        if (sec == 0){
+                            min++;
+                            if (min > 59){
+                                min = 0;
+                                hour++;
+                            }
+                        }
+                        if (sec < 10) {
+                            if (min < 10) {
+                                if (hour < 10) {
+                                    mRuntime.setText("0" + hour + ":0" + min + ":0" + sec);
+                                } else {
+                                    mRuntime.setText("" + hour + ":0" + min + ":0" + sec);
+                                }
+                            } else {
+                                if (hour < 10) {
+                                    mRuntime.setText("0" + hour + ":" + min + ":0" + sec);
+                                } else {
+                                    mRuntime.setText("" + hour + ":" + min + ":0" + sec);
+                                }
+                            }
+                        } else {
+                            if (min < 10) {
+                                if (hour < 10) {
+                                    mRuntime.setText("0" + hour + ":0" + min + ":" + sec);
+                                } else {
+                                    mRuntime.setText("" + hour + ":0" + min + ":" + sec);
+                                }
+                            } else {
+                                if (hour < 10) {
+                                    mRuntime.setText("0" + hour + ":" + min + ":" + sec);
+                                } else {
+                                    mRuntime.setText("" + hour + ":" + min + ":" + sec);
+                                }
+                            }
+                        }
```

Comment:
These if statements can be simplified:
```suggestion
                        mRuntime.setText(String.format("%-2s:%-2s:%-2s", hour, min, sec).replace(' ', '0'));
```

Reply:
I have added a cleaner way of calculating the Date in  therefore this is not needed anymore: https://github.com/twireapp/Twire/pull/211/commits/bafd1f03537c69aa5f7ba141d6d3d7a4fe953569